units/src/amount/signed.rs

```diff
@@ -57,14 +57,6 @@ mod encapsulate {
         /// The minimum value of an amount.
         pub const MIN: Self = Self(-21_000_000 * 100_000_000);
 
-        /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
-        ///
-        /// Accepts an `i32` which is guaranteed to be in range for the type, but which can only
-        /// represent roughly -21.47 to 21.47 BTC.
-        pub const fn from_sat_i32(satoshi: i32) -> SignedAmount {
-            SignedAmount(satoshi as i64) // cannot use i64::from in a constfn
-        }
```

Comment:
The reason for this is not entirely obvious but the change is correct. So to clarify: this being inside privacy boundary allows it to not have the "runtime" check (most likely optimized-away after inlining). But if we wanted to get rid of that check we should have `_unchecked` method instead. But we don't want that (yet), since the check here will have zero performance impact in optimized builds and it's not worth the cost of dealing with unchecked constructors to optimize debug builds.

Since you have to rebase anyway, please add it to the commit message or code comment.

Reply:
Added to commit log.

Reply:
Thanks for more explanation. is my understanding correct that:
* Items inside encapsulate are in a privacy boundary?
* Items inside encapsulate are _not_ "runtime" checked.
* moving from_sat_i32 outside of encapsulate means it is "runtime" checked?
* Items inside the privacy boundary access the inner-field while those outside the privacy boundary do not?
 
Wouldn't `mod private` be more descriptive than `mod encapsulate` if items inside encapsulate are insides a "privacy boundary"?  Furthermore, I'm struggling with the notion of "constructors" that are in a "privacy boundary".  Looking at these concepts from the outside of those developing has left me confused, although I'm sure to those implementing it it's perfectly logical.

Reply:
I see now what the motivation is here, and that `SignedAmount(satoshi as i64)` is only usable inside the `encapulate` namespace.  Reducing the reliance on the inner field would make changes and refactor less noisy.

Reply:
There's also the part about not breaking invariant by accident. Basically, to make sure nothing breaks the invariant we only have to audit the submodule and nothing else. That makes code review much easier.